### PR TITLE
Feature/3166 fix delete extension on blueprints screen

### LINKED ISSUE #3166: Unable to delete extension on blueprints screen
Reported by @orliesaurus: https://pixiebrix.slack.com/archives/C02CN01JXAA/p1649735959899869

Looking at the code, the delete extension handler is wrapped in useUserAction, so we should at least be seeing an error or success/toast: http://github.com/pixiebrix/pixiebrix-extension/blob/87d9c266fa06a74e512ead5c78e6a2bec00d6652/src/options/pages/blueprints/useInstallableActions.ts#L109-L109

Most likely there's an inconsistency here: http://github.com/pixiebrix/pixiebrix-extension/blob/87d9c266fa06a74e512ead5c78e6a2bec00d6652/src/options/pages/blueprints/BlueprintActions.tsx#L112-L112. I think we should change the `hide` property to be whether or not deleteExtension is defined?

### PR DESCRIPTION
**Pull request by mnholtz**
Fixes #3166 by hiding the delete action for installables that are not cloud extensions.

**Open question**
Obviously, Orlando wanted to delete this resource. But way things are currently, he can't. Should we add a feature that allows developers to delete shared blueprints?

### CODE REVIEW

**Comment by twschiller** [comment]
> Should we add a feature that allows developers to delete shared blueprints?

Currently users can delete them from the Workshop. But yes, in the future we may want to introduce a way to delete them without having to go into the workshop. Could you write a ticket for this for us to address in the future?

File: src/options/pages/blueprints/useInstallableViewItems.tsx [hunk 1]
```diff
@@ -50,9 +50,9 @@ function useInstallableViewItems(installables: Installable[]): {
 } {
   const scope = useSelector(selectScope);
   const installedExtensions = useSelector(selectExtensions);
-  const organizations = useGetOrganizationsQuery();
-  const listings = useGetMarketplaceListingsQuery();
-  const recipes = useGetRecipesQuery();
+  const organizationsQuery = useGetOrganizationsQuery();
+  const listingsQuery = useGetMarketplaceListingsQuery();
+  const recipesQuery = useGetRecipesQuery();
```
Comment (mnholtz) [inline src/options/pages/blueprints/useInstallableViewItems.tsx:55]: Unrelated refactoring change for better legibility. This came up while pairing with Ben.

**Comment by mnholtz** [comment]
> Could you write a ticket for this for us to address in the future?

https://github.com/pixiebrix/pixiebrix-extension/issues/3169

File: src/options/pages/blueprints/useInstallableViewItemActions.ts [hunk 2]
```diff
@@ -200,18 +210,26 @@ function useInstallableActions(installable: Installable) {
...
     requestPermissions: hasPermissions ? null : requestPermissions,
     reinstall:
-      isExtensionFromRecipe(installable) || isBlueprint(installable)
+      (isExtensionFromRecipe(installable) || isBlueprint(installable)) &&
+      // Managed extensions are updated via the deployment banner
+      sharing.source.type !== "Deployment" &&
+      status === "Active"
         ? reinstall
         : null,
   };
```
Comment (BALEHOK) [inline src/options/pages/blueprints/useInstallableViewItemActions.ts:232]: I like how the logic is handled in the hook and the UI part of the actions (like icons) is still in the component.

How about tests for the hook?

**Review by BALEHOK — APPROVED** [review]
LGTM. Would be great to have tests for the Actions logic.

File: src/options/pages/blueprints/useInstallableViewItemActions.ts (hunk 2 above)
Comment (mnholtz) [reply src/options/pages/blueprints/useInstallableViewItemActions.ts]: Sweet, thanks aleksandr! [Ended up writing some tests.](https://github.com/pixiebrix/pixiebrix-extension/blob/a1df5605d2981dc60fcf60a4dc9912f2553bf710/src/options/pages/blueprints/useInstallableViewItemActions.test.tsx#L61-L61)

**Pull request merged**